service/Core/Handlers/SaveRecordsHandler.cs

```diff
@@ -273,6 +287,11 @@ private static IEnumerable<FileDetailsWithRecordId> GetListOfPartitionAndSynthet
             .Select(x => new FileDetailsWithRecordId(pipeline, x.Value)));
     }
 
+    private static string GetPipelineKeyForSave(IMemoryDb client, string index)
```

Comment:
The function name is a bit odd... 

It's not clear why the code uses the client class name, could you add a comment? I would consider using `ConditionalWeakTable` so the code will scale to multiple instances of the same type (connected to different endpoints)

Reply:
added TODO for future wok